tracing/otel.go

```diff
@@ -0,0 +1,253 @@
+	envJaegerTags                      = "JAEGER_TAGS"
+	envJaegerSamplerManagerHostPort    = "JAEGER_SAMPLER_MANAGER_HOST_PORT"
+	envJaegerSamplerParam              = "JAEGER_SAMPLER_PARAM"
+	envJaegerEndpoint                  = "JAEGER_ENDPOINT"
+	envJaegerAgentPort                 = "JAEGER_AGENT_PORT"
+	envJaegerSamplerType               = "JAEGER_SAMPLER_TYPE"
+	envJaegerSamplingEndpoint          = "JAEGER_SAMPLING_ENDPOINT"
+	envJaegerDefaultSamplingServerPort = 5778
+	envJaegerDefaultUDPSpanServerHost  = "localhost"
+	envJaegerDefaultUDPSpanServerPort  = "6831"
+)
+
+// NewOTelFromJaegerEnv is a convenience function to allow OTel tracing configuration via Jaeger environment variables
+//
+// Tracing will be enabled if one (or more) of the following environment variables is used to configure trace reporting:
+// - JAEGER_AGENT_HOST
+// - JAEGER_SAMPLER_MANAGER_HOST_PORT
+func NewOTelFromJaegerEnv(serviceName string) (io.Closer, error) {
+	cfg, err := parseTracingConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not load jaeger tracer configuration")
+	}
+	if cfg.samplingServerURL == "" && cfg.agentHostPort == "" && cfg.collectorEndpoint == "" {
+		return nil, ErrBlankTraceConfiguration
+	}
+	return cfg.initJaegerTracerProvider(serviceName)
+}
+
+// parseJaegerTags Parse Jaeger tags from env var JAEGER_TAGS, example of TAGs format: key1=value1,key2=${value2:value3} where value2 is an env var
+// and value3 is the default value, which is optional.
+func parseJaegerTags(sTags string) ([]attribute.KeyValue, error) {
+	pairs := strings.Split(sTags, ",")
+	res := make([]attribute.KeyValue, 0, len(pairs))
+	for _, p := range pairs {
+		k, v, found := strings.Cut(p, "=")
+		if found {
+			k, v := strings.TrimSpace(k), strings.TrimSpace(v)
+			if strings.HasPrefix(v, "${") && strings.HasSuffix(v, "}") {
+				e, d, _ := strings.Cut(v[2:len(v)-1], ":")
+				v = os.Getenv(e)
+				if v == "" && d != "" {
+					v = d
+				}
+			}
+			if v == "" {
+				return nil, errors.Errorf("invalid tag %q, expected key=value", p)
+			}
+			res = append(res, attribute.String(k, v))
+		} else if p != "" {
+			return nil, errors.Errorf("invalid tag %q, expected key=value", p)
+		}
+	}
+	return res, nil
+}
+
+type config struct {
+	agentHost         string
+	collectorEndpoint string
+	agentPort         string
+	samplerType       string
+	samplingServerURL string
+	samplerParam      float64
+	jaegerTags        []attribute.KeyValue
+	agentHostPort     string
+}
+
+// parseTracingConfig facilitates initialization that is compatible with Jaeger's InitGlobalTracer method.
+func parseTracingConfig() (config, error) {
+	cfg := config{}
+	var err error
+
+	// Parse reporting agent configuration
+	if e := os.Getenv(envJaegerEndpoint); e != "" {
+		u, err := url.ParseRequestURI(e)
+		if err != nil {
+			return cfg, errors.Wrapf(err, "cannot parse env var %s=%s", envJaegerEndpoint, e)
+		}
+		cfg.collectorEndpoint = u.String()
```

Comment:
Kind of weird to call it collectorEndpoint as we're using Jaeger exporter to write to it and also doesn't come from OTEL env vars: https://opentelemetry.io/docs/specs/otel/protocol/exporter/#endpoint-urls-for-otlphttp

Let's call it cfg.jaegerEndpoint

Reply:
Fair enough, renamed in https://github.com/grafana/dskit/pull/681/commits/4243e6da74529685c531cb27c80ae411b1f77f1a